packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -106,151 +289,141 @@ function WorklistDataGrid(props) {
+            nextRow = WorklistDataGridUtils.FIRST_NON_HEADER_ROW;
+          }
+        } else {
+          // Left key
+          nextCol -= 1;
+        }
         break;
       case KeyCode.KEY_RIGHT:
-        nextCol += 1;
+        if (event.metaKey) {
+          // Cmd + Right = End
+          nextCol = displayedColumns.length - 1;
+
+          if (event.ctrlKey) {
+            // Ctrl + Cmd + Right = Ctrl + End
+            nextRow = rows.length;
+          }
+        } else {
+          // Right key
+          nextCol += 1;
+        }
         break;
       case KeyCode.KEY_HOME:
         nextCol = 0;
         if (event.ctrlKey) {
-          nextRow = 1; // Assumption is that the first row is the column Heading.
+          nextRow = WorklistDataGridUtils.FIRST_NON_HEADER_ROW; // Assumption is that the first row is the column Heading.
         }
         break;
       case KeyCode.KEY_END:
-        nextCol = props.rows[0].cells.length - 1;
+        nextCol = displayedColumns.length - 1; // Col are zero based.
         if (event.ctrlKey) {
-          nextRow = props.rows.length;
+          // Though rows are zero based, the header is the first row so the rowsLength will
+          // always be one more than then actual number of data rows.
+          nextRow = rows.length;
         }
         break;
+      case KeyCode.KEY_ESCAPE:
+        if (!hasSelectableRows) {
+          handleCellSelectionChange(null, null, cellCoordinates);
+        } else {
+          handleClearRowSelection();
+        }
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_A:
+        if (hasSelectableRows && (event.ctrlKey || event.metaKey)) {
+          selectRows(true, null, null);
+          event.preventDefault(); // prevent the default selection of everything on the page.
+        }
+        return;
       default:
         return;
     }
-
-    if (nextRow >= grid.current.rows.length || nextCol >= grid.current.rows[0].cells.length) {
-      event.preventDefault();
+    if (nextRow > rows.length || nextCol >= displayedColumns.length) {
+      event.preventDefault(); // prevent the page from moving with the arrow keys.
       return;
     }
     if (nextCol < 0 || nextRow < 0) {
-      event.preventDefault();
+      event.preventDefault(); // prevent the page from moving with the arrow keys.
       return;
     }
-
-    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
-    // This also ensures that tabbing back into the grid focuses on the cell that last had focus.
-    grid.current.rows[currentFocusedRow].cells[currentFocusedCol].tabIndex = -1;
-
-    focusedRow.current = nextRow;
-    focusedCol.current = nextCol;
-
-    const elementWithTabStop = grid.current.rows[nextRow].cells[nextCol];
-    elementWithTabStop.tabIndex = 0;
-    elementWithTabStop.focus();
-    event.preventDefault();
+    handleMoveCellFocus(cellCoordinates, { row: nextRow, col: nextCol });
```

Comment:
Doesn't the setFocusedRowCol method do all of this for you?  Is this handleMoveCellFocus method necessary?

Reply:
They are slightly different. Since handleMoveCellFocus knows both the origin and destination, it cleans up the origin and then calls setFocusedRowCol to make the changes for the destination. It may be possible to not use the origin and rely on the focusedRow and focusedCol but in this case we know the origin so we use it.